packages/web-components/lib/components/selfie/src/smartselfie-capture/hooks/useCamera.ts

```diff
@@ -39,6 +64,31 @@ export const useCamera = () => {
       if (videoRef.current) {
         videoRef.current.srcObject = stream;
         await videoRef.current.play();
+
+        if (isSwitchingCameraRef.current && onCameraSwitchCallbackRef.current) {
+          // wait for video to be ready and call callback
+          const video = videoRef.current;
+          const triggerCallback = () => {
+            if (video.videoWidth > 0 && video.videoHeight > 0) {
+              setTimeout(() => {
+                onCameraSwitchCallbackRef.current?.();
+                isSwitchingCameraRef.current = false;
+              }, 100);
+            } else {
+              // if not ready, wait for loadedmetadata
+              const handleReady = () => {
+                video.removeEventListener('loadedmetadata', handleReady);
+                setTimeout(() => {
+                  onCameraSwitchCallbackRef.current?.();
+                  isSwitchingCameraRef.current = false;
+                }, 100);
+              };
+              video.addEventListener('loadedmetadata', handleReady);
+            }
```

Comment:
Can the callbacks and timeouts be cleared so that we can prevent/minimise memory leaks?

Reply:
This might also introduce race conditions.
The  onCameraSwitch callback and isSwitching of can be moved to and Effect
```js
useEffect(() => {
    const video = videoRef.current;

    const handleVideoPlaying = () => {
      if (isSwitchingCameraRef.current) {
        onCameraSwitchCallbackRef.current?.();
        isSwitchingCameraRef.current = false;
      }
    };

      video?.addEventListener('playing', handleVideoPlaying);
    
    return () => {
        video?.removeEventListener('playing', handleVideoPlaying);
    };
  }, [videoRef.current]);
```